test/jira-tests.js

```diff
@@ -40,6 +41,24 @@ describe('Jira API Tests', () => {
       expect(jira.baseOptions.auth).to.be.undefined;
     });
 
+    it('Constructor with oauth credentials', () => {
+      const options = getOptions({ oauth: {
+        consumer_key: 'consumer',
+        consumer_secret: 'consumer_secret',
+        access_token: 'token',
+        access_token_secret: 'token_secret'
+      } });
+
+      const jira = new JiraApi(options);
+
+      expect(jira.baseOptions.auth).to.be.undefined;
+      expect(jira.baseOptions.oauth.consumer_key).to.eql('consumer');
+      expect(jira.baseOptions.oauth.consumer_secret).to.eql('consumer_secret');
+      expect(jira.baseOptions.oauth.token).to.eql('token');
+      expect(jira.baseOptions.oauth.token_secret).to.eql('token_secret');
+      expect(jira.baseOptions.oauth.signature_method).to.eql('RSA-SHA1');
```

Comment:
In the event that this test doesn't pass, let's say because `jira.baseOptions.oauth.token` is not the expected value of `'token'` and because `jira.baseOptions.oauth.token_secret` is not the expected value of `'token_secret'`, you would have to run this test twice to find that out.  The test will fail on the first `expect` failure, so the first time would break on the expect around token, and then the second time would break around token_secret.

It is preferrable to do your expect statement as follows:
```javascript
expect(jira.baseOptions.oauth).to.eql({
  consumer_key: 'consumer',
  consumer_secret: 'consumer_secret',
  token: 'token',
  token_secret: 'token_secret',
  signature_method: 'RSA-SHA1'
});
```
This way, if `jira.baseOptions.oauth.signature_method` is not the whole object we expect, the testing framework will show us every property that is missing, different, the same, or extra.